h/celery.py

```diff
@@ -108,3 +108,39 @@ def start(argv, bootstrap):
     # instance, and it's then used in the worker bootstrap subscriber above.
     celery.webapp_bootstrap = bootstrap
     celery.start(argv)
+
+
+@signals.task_prerun.connect
+def add_task_name_and_id_to_log_messages(
+    task_id, task, *args, **kwargs
+):  # pragma: no cover
+    """
+    Add the Celery task name and ID to all messages logged by Celery tasks.
+
+    This makes it easier to observe Celery tasks by reading the logs. For
+    example you can find all messages logged by a given Celery task by
+    searching for the task's name in the logs.
+
+    This affects:
+
+    * Messages logged by Celery itself
+
+    * Messages logged by our @celery.task functions, perhaps using Celery's
+      get_task_logger()
+
+    * Messages logged by other parts of our code that get called (directly or
+      indirectly) by @celery.task functions and that use a normal logger (e.g.
+      logging.getLogger(__name__))
+
+    * Messages logged by third-party libraries that get called (directly or
+      indirectly) by @celery.task functions and that use their own loggers
+      (e.g. the elasticsearch library uses a logger called "elasticsearch")
+    """
+
+    logging.getLogger().handlers[0].setFormatter(
+        logging.Formatter(
+            "[%(asctime)s: %(levelname)s/%(processName)s] "
+            + f"{task.name}[{task_id}] "
+            + "%(message)s"
```

Comment:
This is the same as the default format (or whatever format was being used before this, anyway) but with the `f"{task.name}[{task_id}] "` inserted.

The way this works is by getting the root logger of the Celery worker process (`logging.getLogger()`) and replacing its formatter. We're assuming that the root logger has exactly one handler (`logging.getLogger().handlers[0]`), which happens to be the case.

Reply:
Any chance of a comment here? Or assign `root_logger = logging.getLogger().handlers[0]` or something? If we feel the need to explain this it's a good hint it's not obvious

Reply:
Changed this to:

```python
    # Replace the root logger's formatter with one that includes task.name and
    # task_id in the format. This assumes that the root logger has one handler,
    # which happens to be the case.
    root_loggers_handler = logging.getLogger().handlers[0]

    root_loggers_handler.setFormatter(
        logging.Formatter(
            "[%(asctime)s: %(levelname)s/%(processName)s] "
            + f"{task.name}[{task_id}] "
            + "%(message)s"
        )
    )
```